lib/serverBuilder.js

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright (c) 2018-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+/* eslint-disable complexity */
+/* eslint-disable max-statements */
+
+require('./vendor/polyfills');
+
+var util              = require('./util');
+var tx                = require('./tx');
+var storage           = require('./serverStorage');
+var AuthSdkError      = require('./errors/AuthSdkError');
+var config            = require('./config');
+
+function OktaAuthBuilder(args) {
+  var sdk = this;
+
+  if (!args) {
+    throw new AuthSdkError('No arguments passed to constructor. ' +
+      'Required usage: new OktaAuth(args)');
+  }
+
+  var url = args.url;
+  if (!url) {
+    throw new AuthSdkError('No url passed to constructor. ' +
+    'Required usage: new OktaAuth({url: "https://sample.okta.com"})');
+  }
+
+  if (url.indexOf('-admin.') !== -1) {
+    throw new AuthSdkError('URL passed to constructor contains "-admin" in subdomain. ' +
+      'Required usage: new OktaAuth({url: "https://dev-12345.okta.com})');
```

Comment:
We've adopted a common pattern of `https://{yourOktaDomain}`

See https://oktawiki.atlassian.net/wiki/spaces/UX/pages/470681125/Developer+Experience+-+Okta+Domain+Labels